src/main/java/de/aaschmid/gradle/plugins/cpd/CpdPlugin.java

```diff
@@ -74,6 +75,7 @@ public class CpdPlugin implements Plugin<Project> {
     @Override
     public void apply(Project project) {
         project.getPlugins().apply(ReportingBasePlugin.class);
+        project.getPlugins().apply(JvmEcosystemPlugin.class);
```

Comment:
Have you tried the tests without this additional "dependency"? The docs say that it provides sources sets (and others) but this should already be available, shouldn't it?

Reply:
I don't remember why I needed this. I'm unable to use the unreleased code in my own projects (via `--include-build`), is there a way to publish the plugin the local Maven repository?

Reply:
I don't think this is possible anymore since Gradle moved to the `publish-plugin` plugin...

I was just about running all the tests locally without the line, give me some more minutes :-)

Reply:
According to my browser history, I had an issue with Guava and how Gradle handles variants:

"Cannot choose between the following variants of com.google.guava:guava:33.0.0-jre:"

Reply:
Without it, 24 tests fail with 

```
result.output.contains("BUILD SUCCESSFUL")
|      |      |
|      |      false
|      > Task :cpdCheck FAILED
|       
|      FAILURE: Build failed with an exception.
|       
|      * What went wrong:
|      Execution failed for task ':cpdCheck'.
|      > Could not resolve all files for configuration ':cpd'.
|         > Could not resolve com.google.guava:guava:33.0.0-jre.
|           Required by:
|               project : > net.sourceforge.pmd:pmd-dist:7.2.0 > net.sourceforge.pmd:pmd-languages-deps:7.2.0 > net.sourceforge.pmd:pmd-apex:7.2.0
|            > Cannot choose between the following variants of com.google.guava:guava:33.0.0-jre:
|                - androidRuntimeElements
|                - jreRuntimeElements
|              All of them match the consumer attributes:
|                - Variant 'androidRuntimeElements' capabilities com.google.collections:google-collections:33.0.0-jre and com.google.guava:guava:33.0.0-jre:
|                    - Unmatched attributes:
|                        - Provides org.gradle.category 'library' but the consumer didn't ask for it
|                        - Provides org.gradle.dependency.bundling 'external' but the consumer didn't ask for it
|                        - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
|                        - Provides org.gradle.jvm.version '8' but the consumer didn't ask for it
|                        - Provides org.gradle.libraryelements 'jar' but the consumer didn't ask for it
|                        - Provides org.gradle.status 'release' but the consumer didn't ask for it
|                        - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
|                - Variant 'jreRuntimeElements' capabilities com.google.collections:google-collections:33.0.0-jre and com.google.guava:guava:33.0.0-jre:
|                    - Unmatched attributes:
|                        - Provides org.gradle.category 'library' but the consumer didn't ask for it
|                        - Provides org.gradle.dependency.bundling 'external' but the consumer didn't ask for it
|                        - Provides org.gradle.jvm.environment 'standard-jvm' but the consumer didn't ask for it
|                        - Provides org.gradle.jvm.version '8' but the consumer didn't ask for it
|                        - Provides org.gradle.libraryelements 'jar' but the consumer didn't ask for it
|                        - Provides org.gradle.status 'release' but the consumer didn't ask for it
|                        - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
|         > Could not resolve com.google.guava:guava:31.1-jre.
|           Required by:
|               project : > net.sourceforge.pmd:pmd-dist:7.2.0 > net.sourceforge.pmd:pmd-languages-deps:7.2.0 > net.sourceforge.pmd:pmd-apex:7.2.0 > com.google.summit:summit-ast:2.2.0
|            > Cannot choose between the following variants of com.google.guava:guava:33.0.0-jre:
|                - androidRuntimeElements
|                - jreRuntimeElements
|              All of them match the consumer attributes:
|                - Variant 'androidRuntimeElements' capabilities com.google.collections:google-collections:33.0.0-jre and com.google.guava:guava:33.0.0-jre:
|                    - Unmatched attributes:
|                        - Provides org.gradle.category 'library' but the consumer didn't ask for it
|                        - Provides org.gradle.dependency.bundling 'external' but the consumer didn't ask for it
|                        - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
|                        - Provides org.gradle.jvm.version '8' but the consumer didn't ask for it
|                        - Provides org.gradle.libraryelements 'jar' but the consumer didn't ask for it
|                        - Provides org.gradle.status 'release' but the consumer didn't ask for it
|                        - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
|                - Variant 'jreRuntimeElements' capabilities com.google.collections:google-collections:33.0.0-jre and com.google.guava:guava:33.0.0-jre:
|                    - Unmatched attributes:
|                        - Provides org.gradle.category 'library' but the consumer didn't ask for it
|                        - Provides org.gradle.dependency.bundling 'external' but the consumer didn't ask for it
|                        - Provides org.gradle.jvm.environment 'standard-jvm' but the consumer didn't ask for it
|                        - Provides org.gradle.jvm.version '8' but the consumer didn't ask for it
|                        - Provides org.gradle.libraryelements 'jar' but the consumer didn't ask for it
|                        - Provides org.gradle.status 'release' but the consumer didn't ask for it
|                        - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
|       
|      * Try:
|      > Run with --stacktrace option to get the stack trace.
|      > Run with --info or --debug option to get more log output.
|      > Run with --scan to get full insights.
|       
|      * Get more help at https://help.gradle.org
|       
|      BUILD FAILED in 44ms
|      1 actionable task: 1 executed
<org.gradle.testkit.runner.internal.FeatureCheckBuildResult@7fc4e652 delegateBuildResult=org.gradle.testkit.runner.internal.DefaultBuildResult@50fe4d62 outputFeatureCheck=org.gradle.testkit.runner.internal.feature.BuildResultOutputFeatureCheck@1f25c60>

	at de.aaschmid.gradle.plugins.cpd.test.CpdIntegrationTest.executing 'Cpd' task on duplicate 'kotlin' source should not fail if 'ignoreFailures' and produce 'cpdCheck.csv' with one warning(CpdIntegrationTest.groovy:180)
```
So I would suggest just keeping it ;-)

Reply:
Interesting, I wasn't able to reproduce this today. But yeah, I remember having that issue.

Reply:
Ok, so lets just keep it.